renku/core/commands/checks/githooks.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Check for required Git hooks."""
+from io import StringIO
+from pathlib import Path
+
+import pkg_resources
+from git.index.fun import hook_path as get_hook_path
+
+from renku.core.management.githooks import HOOKS
+
+from ..echo import WARNING
+
+
+def check_git_hooks_installed(client):
+    """Checks if all necessary hooks are installed."""
+    for hook in HOOKS:
+        hook_path = Path(get_hook_path(hook, client.repo.git_dir))
+        if not hook_path.exists():
+            message = WARNING + 'Git hooks are not installed. ' \
+                'Use "renku githooks install" to install them. \n'
+            return False, message
+
+        with hook_path.open() as file_:
+            actual_hook = _extract_renku_hook(file_)
+        with StringIO(_read_resource(hook)) as file_:
+            expected_hook = _extract_renku_hook(file_)
+
+        if not expected_hook:
+            message = WARNING + 'Cannot check for existence of Git hooks.\n'
```

Comment:
I guess this would happen if the renku installation is somehow corrupted? Maybe we should suggest to reinstall renku?

Reply:
This was for myself to avoid comparing empty dicts in case the pattern was not parsed correctly :D. This is basically a safety-net for us in case we update the hooks and mess something up so that we see some tests fail. User should never see this and if something is wrong with their renku installation then they probably won't even reach this code.